### ISSUE TITLE
Panic: runtime error

### ISSUE THREAD

**Issue by edoardottt** [hahwul/dalfox#216]
## Describe the bug
```
panic: runtime error: index out of range [2] with length 2

goroutine 1 [running]:
github.com/hahwul/dalfox/v2/pkg/scanning.Scan(0xc00002c100, 0x34, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/edoardottt/go/pkg/mod/github.com/hahwul/dalfox/v2@v2.3.6/pkg/scanning/scan.go:197 +0x574e
github.com/hahwul/dalfox/v2/cmd.glob..func2(0x19adbe0, 0xc000126620, 0x0, 0x2)
	/home/edoardottt/go/pkg/mod/github.com/hahwul/dalfox/v2@v2.3.6/cmd/pipe.go:99 +0xb38
github.com/spf13/cobra.(*Command).execute(0x19adbe0, 0xc000126600, 0x2, 0x2, 0x19adbe0, 0xc000126600)
	/home/edoardottt/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0x19ade60, 0x443cfa, 0x106d940, 0xc000000180)
	/home/edoardottt/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/home/edoardottt/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
github.com/hahwul/dalfox/v2/cmd.Execute()
	/home/edoardottt/go/pkg/mod/github.com/hahwul/dalfox/v2@v2.3.6/cmd/root.go:38 +0x31
main.main()
	/home/edoardottt/go/pkg/mod/github.com/hahwul/dalfox/v2@v2.3.6/dalfox.go:10 +0x20
```

## Environment
 * Dalfox Version: `v2.3.6`
 * Installed from: `GO111MODULE=on go get -v github.com/hahwul/dalfox/v2`

**Comment by hahwul** [comment]
Hi @edoardottt 
First, thank you for submit issue :D
is there any way, could you give me testing command or crash sample? 

----
for me (197 line, crach point)
https://github.com/hahwul/dalfox/blob/1f32f3494e1aa3312f84b3e2a836eb61a9ae9aac/pkg/scanning/scan.go#L197

**Comment by edoardottt** [comment]
Yeah it's absolutely that line causing the problem.
Unfortunately I forgot to save the urls txt file where I was executing dalfox..
I will try to reproduce the steps, in case the error will fire up I will post here (or in private Idk) the file :)

**Comment by hahwul** [comment]
@edoardottt 
Oh, thank you so much! First of all, I am going to apply the defense code, and try to find out the cause :D

**Comment by edoardottt** [comment]
Can I send u a direct message on twitter? I have a question

**Comment by edoardottt** [comment]
- `echo "https://cmm.randstad.com/wp-login.php?redirect_to=/" > poc.txt`
- `cat poc.txt | dalfox pipe`

**Comment by hahwul** [comment]
@edoardottt 
sure :D
and I don't think it's an issue that happens unconditionally 😭. Processing exceptions for now.

![1414](https://user-images.githubusercontent.com/13212227/113309995-b04ea200-9342-11eb-93bf-748a8df2a2fb.jpg)

**Comment by hahwul** [comment]
@edoardottt 
I just added a defense code, so could you please test me for the same issue?

```go
for k, v := range options.PathReflection {
		split := strings.Split(parsedURL.Path, "/")
		if len(split) > k+1 {
			split[k+1] = options.AuroraObject.Yellow("dalfoxpathtest").String()
			tempURL := strings.Join(split, "/")
			printing.DalLog("INFO", "Reflected PATH '"+tempURL+"' => "+v+"]", options)
		}
	}
```

## Test
```
$ git clone https://github.com/hahwul/dalfox
$ cd dalfox
$ go build
```

version check
```
$ ./dalfox verison
v2.3.7-dev
```

Test...
```
$ ./dalfox ~~~
```

**Comment by edoardottt** [comment]
Ok, I will text you when I have more free time, thanks :)

And I confirm this defense code is working perfectly! With the `v2.3.6` it keeps crashing, while the `v.2.3.7-dev` doesn't. I don't know the specific reason actually.. It could be my Ubuntu / Go specific environment..

Thanks hahwul!!

**Comment by hahwul** [comment]
@edoardottt 
Thank you so much for your help in solving the problem.  Maybe... `v2.3.7` will be released soon. I'll give you a comment when it's released.

**Comment by hahwul** [comment]
Closing the issue because released version 2.3.7. If there is any other problem, please re-open it! Enjoy your day!
